application/libraries/Aauth.php

```diff
@@ -121,42 +121,40 @@ public function __construct() {
 		$this->errors = $this->CI->session->flashdata('errors') ?: array();
 		$this->infos = $this->CI->session->flashdata('infos') ?: array();
 
-		// Pre-Cache IDs
-		$this->precache_ids();
-
-	}
-
-	/**
-	 * pre_cache_ids() caches all permission and group IDs for later use.
-	 */
-	public function precache_ids() {
-
 		// Initialize Variables
 
 		$this->cache_perm_id		= array();
 		$this->cache_group_id		= array();
+		
+		// Pre-Cache IDs
+		$this->precache_perms();
+		$this->precache_groups();
 
-		// Permissions
-
+	}
+	
+	/**
+	 * precache_perms() caches all permission IDs for later use.
+	 */
+	private function precache_perms {
```

Comment:
The function name line should be as follows:
private function precache_perms() {

Observer the missing '()' after the function name

Reply:
And the same thing with the function 
precache_groups